connector/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ExecutePlanResponseObserver.scala

```diff
@@ -15,27 +15,32 @@
  * limitations under the License.
  */
 
-package org.apache.spark.sql.connect.service
+package org.apache.spark.sql.connect.execution
 
-import org.apache.spark.connect.proto
+import io.grpc.stub.StreamObserver
+
+import org.apache.spark.connect.proto.ExecutePlanResponse
 
 /**
- * Object used to hold the Spark Connect execution state.
+ * Container for responses to Execution.
+ * TODO: At this moment, it simply forwards response to the underlying GRPC StreamObserver.
+ *       Detaching execution from the RPC handler, this can be used to store the responses,
+ *       and then send them to different GRPC StreamObservers.
+ *
+ * @param responseObserver
  */
-case class ExecutePlanHolder(
-    operationId: String,
-    sessionHolder: SessionHolder,
-    request: proto.ExecutePlanRequest) {
+class ExecutePlanResponseObserver(responseObserver: StreamObserver[ExecutePlanResponse])
+  extends StreamObserver[ExecutePlanResponse] {
```

Comment:
This is a bit of a stub for now, but wanted to put it in already, so I don't need to move code around in followups.